### DISCUSSION TITLE
Inject script on chat dock

### DISCUSSION THREAD

**Discussion by fnando** [obsproject/obs-studio#5647] [General]
Hi there!

I'd like to extend Chat docks with some JavaScript/CSS (mostly Youtube/Twitch). I was wondering if it's possible to be done with OBS scripting. 

I noticed that OBS [already injects code into chats](https://github.com/obsproject/obs-studio/blob/5df4ffa8813b9659eb7d0f41862fb883d07aa6b6/UI/auth-youtube.cpp#L120-L124), so I guess my last resource would be adding the changes that I want there and recompile the project (although it's definitely suboptimal).

Any ideas?

Cheers!

**Comment by sebastian-s-beckmann** [comment]
Unfortunately, for docks this is not possible without recompiling the project.

**Comment by sukrosono** [comment]
Well when recompiling, I have some objectives that maybe you want to know/help.
When using restream chat app (windows only) I can receive a text to speech-based chat, but now I tend to stream to twitch only.
I would pay to recompile when I get that feature, I mean twitch / yt chat box, which can be sent to TSS or not.
